src/sql-parser/src/ast/defs/expr.rs

```diff
@@ -735,3 +743,42 @@ impl<T: AstInfo> AstDisplay for FunctionArgs<T> {
     }
 }
 impl_display_t!(FunctionArgs);
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
+pub enum IsExprConstruct {
+    Null,
+    True,
+    False,
+    Unknown,
+}
+
+impl IsExprConstruct {
+    pub fn requires_boolean_expr(&self) -> bool {
+        match self {
+            IsExprConstruct::Null => false,
+            IsExprConstruct::True | IsExprConstruct::False | IsExprConstruct::Unknown => true,
+        }
+    }
+}
+
+impl fmt::Display for IsExprConstruct {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match self {
+            IsExprConstruct::Null => f.write_str("NULL"),
+            IsExprConstruct::True => f.write_str("TRUE"),
+            IsExprConstruct::False => f.write_str("FALSE"),
+            IsExprConstruct::Unknown => f.write_str("UNKNOWN"),
+        }
+    }
+}
```

Comment:
Oh, this should be auto-derived using the `impl_display!` macro, which just delegates to the `AstDisplay` impl.